src/ZeroLog/LogEvent.cs

```diff
@@ -312,6 +458,44 @@ public void WriteToStringBufferUnformatted(StringBuffer stringBuffer)
                 stringBuffer.Append(LogManager.Config.TruncatedMessageSuffix);
         }
 
+        private void AppendJsonValue(StringBuffer stringBuffer, ref byte* dataPointer)
+        {
+            var argument = *dataPointer;
+            dataPointer += sizeof(ArgumentType);
+
+            var argumentType = (ArgumentType)(argument & ArgumentTypeMask.ArgumentType);
+
+            switch (argumentType)
+            {
+                case ArgumentType.KeyString:
+                    AppendUnformattedArgumentValue(stringBuffer, ArgumentType.String, ref dataPointer);
+                    break;
+                case ArgumentType.Guid:
+                case ArgumentType.DateTime:
+                case ArgumentType.TimeSpan:
+                    stringBuffer.Append('"');
+                    AppendUnformattedArgumentValue(stringBuffer, argumentType, ref dataPointer);
+                    stringBuffer.Append('"');
+                    break;
+                case ArgumentType.Char:
+                    stringBuffer.Append('"');
+                    stringBuffer.Append(*(char*)dataPointer);
+                    stringBuffer.Append('"');
+                    dataPointer += sizeof(char);
+                    break;
+                case ArgumentType.Boolean:
+                    stringBuffer.Append(*(bool*)dataPointer ? "\"true\"" : "\"false\"");
+                    dataPointer += sizeof(bool);
+                    break;
+                case ArgumentType.Null:
+                    stringBuffer.Append("\"null\"");
+                    break;
+                default:
+                    AppendUnformattedArgumentValue(stringBuffer, argumentType, ref dataPointer);
```

Comment:
Same remark as above - this is not suitable here.

Reply:
Fixed, thanks!